packages/core-mobile/app/new/features/browser/components/DiscoverEcosystemProjects.tsx

```diff
@@ -0,0 +1,97 @@
+import { useQuery } from '@tanstack/react-query'
+import { LoadingState } from 'common/components/LoadingState'
+import React, { ReactNode, useCallback, useMemo } from 'react'
+import { FlatList, ListRenderItem, View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { addHistoryForActiveTab, selectIsTabEmpty } from 'store/browser'
+import { useBrowserContext } from '../BrowserContext'
+import { HORIZONTAL_MARGIN } from '../consts'
+import {
+  ContentfulEcosystemProject,
+  fetchEcosystemProjects
+} from '../hooks/useContentful'
+import { CarouselItem } from './CarouselItem'
+
+export const DiscoverEcosystemProjects = (): ReactNode => {
+  const dispatch = useDispatch()
+  const { handleUrlSubmit } = useBrowserContext()
+  const showEmptyTab = useSelector(selectIsTabEmpty)
+
+  const { data, error } = useQuery({
+    queryKey: ['discover-ecosystem-projects'],
+    queryFn: fetchEcosystemProjects
+  })
```

Comment:
we don't really call useQuery directly. could you create hooks for these and use them in the components instead?